### PR TITLE
gitlab modules: unify authentication method

### LINKED ISSUE #64770: gitlab modules : user/password method is deprecated
##### SUMMARY

`python-gitlab` library release a new version [v1.13.0](https://github.com/python-gitlab/python-gitlab/releases/tag/v1.13.0). User/Password authentication as been removed completely causing all Ansible GitLab module to fail... 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

gitlab_group
gitlab_deploy_key
gitlab_hook
gitlab_project
gitlab_project_variable
gitlab_runner
gitlab_user

##### ANSIBLE VERSION

All versions

### PR DESCRIPTION
**Pull request by Lunik**
##### SUMMARY

- Unify the authentication to GitLab in `module_utils`
- Fixes #64770

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

gitlab_group
gitlab_deploy_key
gitlab_hook
gitlab_project
gitlab_project_variable
gitlab_runner
gitlab_user

module_utils/gitlab

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Lunik** [comment]
!needs_revision
ready_for_review
-label WIP

**Review by markuman — APPROVED** [review]
LGTM

straightforward. Integration test of ` gitlab_project_variable`  succeeded.

File: lib/ansible/module_utils/gitlab.py [hunk 1]
```diff
@@ -59,3 +72,33 @@ def findGroup(gitlab_instance, identifier):
...
+    gitlab_user = module.params['api_username']
+    gitlab_password = module.params['api_password']
+    gitlab_token = module.params['api_token']
+
+    if not HAS_GITLAB_PACKAGE:
+        module.fail_json(msg=missing_required_lib("python-gitlab"), exception=GITLAB_IMP_ERR)
+
+    try:
+        # python-gitlab library remove support for username/password authentication since 1.13.0
+        # Changelog : https://github.com/python-gitlab/python-gitlab/releases/tag/v1.13.0
```
Comment (Akasurde) [inline lib/ansible/module_utils/gitlab.py:89]: Small nit-pick. Can you please add changelog entry and porting guide note ? Thanks.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/gitlab.py (hunk 1 above)
Comment (Lunik) [reply lib/ansible/module_utils/gitlab.py:89]: Just added the changelog.
I this case, there is no need for porting guide because it's only a bugfix

**Comment by Lunik** [comment]
ready_for_review

**Pull request merged**